src/main/java/spoon/support/compiler/jdt/JDTImportBuilder.java

```diff
@@ -78,7 +78,9 @@ public void build() {
 
 					if (ctPackage != null) {
 						for (CtType type : ctPackage.getTypes()) {
-							this.imports.add(type.getReference());
+							if (type.getVisibility() == ModifierKind.PUBLIC) {
```

Comment:
should also work with package-visible if both classes are in the same package?

Reply:
Yes, but if both classes are in the same packages, it's useless to import them...